sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/model/SqlStmt.kt

```diff
@@ -263,11 +264,21 @@ class SqlStmt private constructor(
       method.addParameter(parameter.build())
     }
 
+    var argsCodeBlock = CodeBlock.of("new String[0], ")
+
     if (arguments.isNotEmpty()) {
-      // Method body begins with local vars used during computation.
-      method.addStatement("\$T<String> args = new \$T<String>()", LIST_TYPE, ARRAYLIST_TYPE)
-          .addStatement("int currentIndex = 1")
-          .addStatement("\$1T query = new \$1T()", STRINGBUILDER_TYPE)
+      val hasNoStrings = arguments.map { it.argumentType.comparable?.dataType != TEXT }
```

Comment:
this can be `val hasNoStrings = arguments.none { it.argumentType.comparable?.dataType == TEXT }`

and then you can probably just inline it in the if statement as `if (arguments.any { it.argumentType.comparable?.dataType == TEXT }) {`